src/libs/ReportUtils.ts

```diff
@@ -1962,6 +1962,25 @@ function getDisplayNameForParticipant(accountID?: number, shouldUseShortForm = f
 function getParticipantsAccountIDsForDisplay(report: OnyxEntry<Report>, shouldExcludeHidden = false, shouldExcludeDeleted = false): number[] {
     let participantsEntries = Object.entries(report?.participants ?? {});
 
+    // We should not show participants that have an optimistic entry with the same login in the personal details
+    const nonOptimisticLoginMap: Record<string, boolean | undefined> = {};
+
+    for (const entry of participantsEntries) {
+        const [accountID] = entry;
+        const personalDetail = allPersonalDetails?.[accountID];
+        if (personalDetail?.login && !personalDetail.isOptimisticPersonalDetail) {
+            nonOptimisticLoginMap[personalDetail.login] = true;
+        }
+    }
+
+    participantsEntries = participantsEntries.filter(([accountID]) => {
+        const personalDetail = allPersonalDetails?.[accountID];
+        if (personalDetail?.login && personalDetail.isOptimisticPersonalDetail) {
+            return !nonOptimisticLoginMap[personalDetail.login];
+        }
+        return true;
+    });
```

Comment:
Isn't that what the code I wrote is doing since it has `!allPersonalDetails?.[accountID]`?

Reply:
filter keeps the entry if you  return true and removes it if you return false, so if it did not have personal details, we would return true and thus not filter it.

Reply:
Because if I am understanding that sentence, the code does that, no? It filters out `participants` which have an entry in `allPersonalDetails` that is optimistic

Reply:
> participantEntries could contain 2 participant records sharing the same login value

Am In looking at this wrong or `participantEntries` (which is really `report.participants`) does not contain any login? 😕